### PR TITLE
feat(packages/rich-list-api-client): `rich-list-api-client`

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What this PR does / why we need it:
Client side library to retrieve rich list

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/rich-list-api-client/src/RichListApiClient.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * RichListApiClient configurable options
+ */
+export interface RichListApiClientOptions extends OceanApiClientOptions {}
+
+/**
+ * RichListApiClient
+ */
+export class RichListApiClient extends OceanApiClient {
+  constructor (
+    protected readonly options: RichListApiClientOptions
```
Comment (canonbrother) [inline packages/rich-list-api-client/src/RichListApiClient.ts:14]:
```suggestion
    options: RichListApiClientOptions
```

File: packages/rich-list-api-client/__tests__/RichListApiClient.spec.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import nock from 'nock'
+import { RichListApiClient } from '../src/RichListApiClient'
+
+const client = new RichListApiClient({
+  url: 'http://rich-list-api-test.internal',
+  network: 'unit-test',
+  version: 'v0',
+  timeout: 60000
```
Comment (canonbrother) [inline packages/rich-list-api-client/__tests__/RichListApiClient.spec.test.ts:8]:
```suggestion
```

Comment (ivan-zynesis) [reply packages/rich-list-api-client/__tests__/RichListApiClient.spec.test.ts:8]: Was wondering how come it get error.

**Review by canonbrother — APPROVED** [review]

File: packages/rich-list-api-client/src/RichListApiClient.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    })
+    this.options.url = this.options.url?.replace(/\/$/, '')
+  }
+
+  async get (tokenId: string): Promise<ApiPagedResponse<RichListItem>> {
+    return await this.requestData('GET', `rich-list/${tokenId}`)
+  }
+}
+
+export interface RichListItem {
```
Comment (cwkang1998) [inline packages/rich-list-api-client/src/RichListApiClient.ts:31]: In #1084, you are calling it `AddressBalance` and here you are using `RichListItem`, I understand that this is from the client perspective and thus `RichListItem` does makes a bit more sense. Is this intended?

File: packages/rich-list-api-client/__tests__/RichListApiClient.test.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          data: {
+            bar: ['1', '2']
+          }
+        }
+      })
+
+    const result = await client.requestData('GET', 'foo')
+    await expect(result).toStrictEqual({
+      bar: ['1', '2']
+    })
```
Comment (cwkang1998) [inline packages/rich-list-api-client/__tests__/RichListApiClient.test.ts:26]: Isn't this behaviour already tested in `RichListApiClient.spec.test.ts` ?  I think we don't need to test it again here.

Comment (ivan-zynesis) [reply packages/rich-list-api-client/__tests__/RichListApiClient.test.ts:26]: Oops, was some leftover from debugging

File: packages/rich-list-api-client/src/RichListApiClient.ts (hunk 3 above)
Comment (ivan-zynesis) [reply packages/rich-list-api-client/src/RichListApiClient.ts:31]: Yes, it is intentional.

In RichListCore, it hold ALL addresses with their known balance.
From client perspective, it retrieve only 1000 rows with highest balance.

**Review by kodemon — DISMISSED** [review]

File: packages/rich-list-api-client/src/RichListApiClient.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * RichListApiClient
+ */
+export class RichListApiClient {
+  private readonly oceanApiClient: OceanApiClient
+
+  constructor (private readonly options: OceanApiClientOptions) {
+    this.oceanApiClient = new OceanApiClient(options)
+  }
+
+  async get (tokenId: string): Promise<ApiPagedResponse<RichListItem>> {
```
Comment (canonbrother) [inline packages/rich-list-api-client/src/RichListApiClient.ts:14]:
```suggestion
  async getToken (tokenId: string): Promise<ApiPagedResponse<RichListItem>> {
```

File: packages/rich-list-api-client/src/RichListApiClient.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * RichListApiClient
+ */
+export class RichListApiClient {
+  private readonly oceanApiClient: OceanApiClient
+
+  constructor (private readonly options: OceanApiClientOptions) {
+    this.oceanApiClient = new OceanApiClient(options)
+  }
```
Comment (canonbrother) [inline packages/rich-list-api-client/src/RichListApiClient.ts:12]: hmm... why don't we standardize the design?? :thinking:

**Comment by ivan-zynesis** [comment]
Superseded by #1173

**Pull request closed without merging**